libcontainer/cgroups/fs/memory.go

```diff
@@ -108,28 +121,28 @@ func setMemoryAndSwap(path string, cgroup *configs.Cgroup) error {
 		// for memory and swap memory, so it won't fail because the new
 		// value and the old value don't fit kernel's validation.
 		if memoryUsage.Limit < uint64(cgroup.Resources.MemorySwap) {
```

Comment:
ah i see what's happening here. `currentmemLimit < uint64(memSwap)` with memSwap = `-1` is practically always true because uint64(-1) is basically MaxUint64.

That being said, do you think its clearer to just add `if cgroup.Resources.MemorySwap == -1 || memoryUsage.Limit < uint64(cgroup.Resources.MemorySwap)` so we dont depend on this uint64(-1) casting. I dont really like it.